subsys/shell/shell.c

```diff
@@ -1331,25 +1333,57 @@ void shell_fprintf(const struct shell *shell, enum shell_vt100_color color,
 	k_mutex_unlock(&shell->ctx->wr_mtx);
 }
 
-void shell_hexdump(const struct shell *shell, const u8_t *data, size_t len)
+static void shell_hexdump_line(const struct shell *shell, unsigned int offset,
+			       const u8_t *data, size_t len)
 {
-	int n = 0;
+	int i;
+
+	shell_fprintf(shell, SHELL_NORMAL, "%08X: ", offset);
 
-	while (len--) {
-		if (n % 16 == 0) {
-			shell_fprintf(shell, SHELL_NORMAL, "%08X: ", n);
+	for (i = 0; i < HEXDUMP_BYTES_IN_LINE; i++) {
+		if (i > 0 && !(i % 8)) {
+			shell_fprintf(shell, SHELL_NORMAL, " ");
 		}
 
-		shell_fprintf(shell, SHELL_NORMAL, "%02X ", *data++);
+		if (i < len) {
+			shell_fprintf(shell, SHELL_NORMAL, "%02x ",
+				      data[i] & 0xFF);
+		} else {
+			shell_fprintf(shell, SHELL_NORMAL, "   ");
+		}
+	}
+
+	shell_fprintf(shell, SHELL_NORMAL, "|");
 
-		n++;
-		if (n % 16 == 0) {
-			shell_print(shell, "");
+	for (i = 0; i < HEXDUMP_BYTES_IN_LINE; i++) {
+		if (i > 0 && !(i % 8)) {
+			shell_fprintf(shell, SHELL_NORMAL, " ");
```

Comment:
I'm not sure if this is needed, at least the 'hd' command in Linux doesn't separate the blocks of printable characters. In fact, doing so makes it hard to distinguish between a white space in the input string and a group separator.

Reply:
hexdump doesn't add space. But wireshark for example does. It is a little bit different case, because wireshark is GUI. I have a slight preference to keep that space. Lets wait for others opinions.

Reply:
OK. I'm not particularly bothered about it.

Reply:
At least for my eyes, it is easier to read the output if the chars are grouped to 8 chars, probably because I have so used to reading wireshark output. So +1 from me for adding 1 space.